integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/core/PutMessageService.java

```diff
@@ -1,46 +1,72 @@
 package no.difi.meldingsutveksling.core;
 
+import no.arkivverket.standarder.noark5.arkivmelding.Arkivmelding;
+import no.difi.meldingsutveksling.arkivmelding.ArkivmeldingFactory;
+import no.difi.meldingsutveksling.arkivmelding.ArkivmeldingUtil;
 import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
 import no.difi.meldingsutveksling.domain.MeldingsUtvekslingRuntimeException;
 import no.difi.meldingsutveksling.logging.Audit;
+import no.difi.meldingsutveksling.nextmove.ConversationDirection;
+import no.difi.meldingsutveksling.nextmove.ConversationResourceRepository;
+import no.difi.meldingsutveksling.nextmove.DirectionalConversationResourceRepository;
+import no.difi.meldingsutveksling.nextmove.DpaConversationResource;
+import no.difi.meldingsutveksling.nextmove.message.MessagePersister;
 import no.difi.meldingsutveksling.noarkexchange.PutMessageMarker;
 import no.difi.meldingsutveksling.noarkexchange.PutMessageRequestWrapper;
 import no.difi.meldingsutveksling.noarkexchange.PutMessageResponseFactory;
 import no.difi.meldingsutveksling.noarkexchange.receive.InternalQueue;
 import no.difi.meldingsutveksling.noarkexchange.schema.PutMessageResponseType;
 import no.difi.meldingsutveksling.receipt.ConversationService;
 import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookup;
+import no.difi.meldingsutveksling.serviceregistry.externalmodel.ServiceRecord;
+import org.springframework.beans.factory.ObjectProvider;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import javax.xml.bind.JAXBException;
+import java.io.IOException;
+import java.util.List;
+
+import static no.difi.meldingsutveksling.ServiceIdentifier.DPA;
+import static no.difi.meldingsutveksling.arkivmelding.ArkivmeldingUtil.ARKIVMELDING_XML;
+
 /**
  *
  * @author Nikolai Luthman <nikolai dot luthman at inmeta dot no>
  */
 @Component
-public class EDUCoreService {
+public class PutMessageService {
 
     private final IntegrasjonspunktProperties properties;
     private final EDUCoreSender coreSender;
     private final ServiceRegistryLookup serviceRegistryLookup;
     private final InternalQueue queue;
     private final ConversationService conversationService;
+    private final ArkivmeldingFactory arkivmeldingFactory;
+    private final DirectionalConversationResourceRepository crRepo;
+    private final MessagePersister persister;
 
     @Autowired
-    public EDUCoreService(
+    public PutMessageService(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)